### PR TITLE
#3540 breach modal UI improvements

### LINKED ISSUE #3540: Breach modal UI improvements
## Describe the bug

The breach modals pretty hunchy funchy. The rest of the UI is lookin pretty good. This is letting the team down.

image

There are multiple things wrong here.

- `This one is like someone's signature during an earthquake`
- The text is all cut off
- In general looks about a 2/10!

### To reproduce

internal code

### Expected behaviour

- The chart should look at most like someone was in a magnitude 2 earthquake not a magnitude 20 when they drew it
- The text should ideally be readable
- I would settle for a 5/10

### Proposed Solution

Play with styles a bit, I guess

### Version and device info

- App version: 7.0.0RC-3
- Tablet model:
- OS version:

### Additional context

N/A

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #3540, #3559

## Change summary

Changed the interpolation used for line charts, as the `natural` was able to go back in time.
Tweaked the columns displayed and added a little padding. Maybe a solid 3/10 now? 

Screenshot_2021-02-19-16-20-14-473

Offroad fix - for #3559 which was committed in a previous PR

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Have a look at the breach chart. The example shown was sensor 2 of the dummy data set.

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

### CODE REVIEW

File: src/utilities/modules/vaccines/constants.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const DOMAIN_OFFSET = 3;
 const AXIS_OFFSET = 50;
 const STROKE_WIDTH = 2;
 const STROKE_SIZE = 3;
-const INTERPOLATION = 'natural';
+const INTERPOLATION = 'catmullRom';
```
Comment (joshxg) [inline src/utilities/modules/vaccines/constants.js:10]: Uhhh.. you're a catmullrom

File: src/widgets/SensorHeader/LastSensorDownload.js [hunk 2]
```diff
@@ -91,7 +91,7 @@ LastSensorDownloadComponent.propTypes = {
   isPaused: PropTypes.bool.isRequired,
   lastDownloadFailed: PropTypes.bool.isRequired,
   lastDownloadTime: PropTypes.instanceOf(Date),
-  logDelay: PropTypes.number,
+  logDelay: PropTypes.oneOfType([PropTypes.number, PropTypes.instanceOf(Date)]),
```
Comment (joshxg) [inline src/widgets/SensorHeader/LastSensorDownload.js:94]: fixes https://github.com/openmsupply/mobile/issues/3558

File: src/widgets/modalChildren/BreachDisplay.js [hunk 3]
```diff
@@ -19,12 +19,16 @@ import { AfterInteractions } from '../AfterInteractions';
 
 const Breach = ({ breach, getInfoColumns }) => {
   const { temperatureLogs } = breach;
+  const lineData = temperatureLogs.map(log => {
+    const { timestamp, temperature } = log;
+    return { timestamp, temperature };
+  });
```
Comment (joshxg) [inline src/widgets/modalChildren/BreachDisplay.js:25]: Hm 🤔 confused why this is needed

**Review by joshxg — APPROVED** [review]
Beautiful 😍

File: src/widgets/modalChildren/BreachDisplay.js (hunk 3 above)
Comment (mark-prins) [reply src/widgets/modalChildren/BreachDisplay.js:25]: well spotted!
uh, it's not. I went wandering down that path thinking that I might have to re-sort the data. didn't in the end. I became convinced that it made the chart slightly faster to load, so kept it in 🤷‍♂️

**Pull request merged**